docs/onboarding-guide.md

```diff
@@ -0,0 +1,171 @@
+
+# Onboarding Guide for .NET Deployment Tool
+
+**AWS Deploy Tool for .NET** is an interactive tooling for the .NET CLI and the AWS Toolkit for Visual Studio that helps deploy .NET applications with minimum AWS knowledge, and with the fewest clicks or commands.
+
+## Reference Links
+
+* Repository - https://github.com/aws/aws-dotnet-deploy
+* Public documentation - https://aws.github.io/aws-dotnet-deploy/
+
+## Using the Deployment Tool from Source
+
+* Ensure that all the [pre-requisties](https://aws.github.io/aws-dotnet-deploy/docs/getting-started/pre-requisites/) (including Docker) are installed and your default AWS profile has admin access
+* Clone the repository using `git clone https://github.com/aws/aws-dotnet-deploy.git`
+* Open `AWS.Deploy.sln` in Visual Studio.
+* Ensure that `../src/AWS.Deloy.CLI` is set as the start up project.
+* Go to `AWS.Deploy.CLI debug properties` by clicking the following drop down
+
+![](debug-properties.png)
+* Add a new launch profile called `AWS.Deploy.CLI`
+* In the *Command line arguments* text box add the following (leave other text boxes blank): 
+
+```
+deploy —project-path .../path/to/aws-dotnet-deploy/testapps/WebAppWithDockerFile
+```
+
+*The above command assumes you have a [default] profile configured in your AWS credentials file. If you want to use a different profile, use the `--profile` switch*
+
+* Select the `AWS.Deploy.CLI` launch profile and run the tool
+* Follow the prompts on the screen and perform a deployment using any compatible recommendation.
+* A successful deployment indicates that your environment is set up correctly and you can proceed to debugging the source code.
+
+## Repository Overview
+
+The .NET deployment tool comprises of various top level components.
+
+
+### [AWS.Deploy.CLI](https://github.com/aws/aws-dotnet-deploy/tree/main/src/AWS.Deploy.CLI)
+
+This component controls the UI elements for the .NET deployment tool and serves as the first entry point when running the tool. This is the layer that end users directly interact with and it uses the [System.CommandLine](https://docs.microsoft.com/en-us/dotnet/standard/commandline/) package for the CLI UX.
+
+#### Key Components
+
+* [**Program.cs**](https://github.com/aws/aws-dotnet-deploy/blob/main/src/AWS.Deploy.CLI/Program.cs) - Serves at the entry point to the CLI application.
+* [**CustomServiceCollectionExtension**](https://github.com/aws/aws-dotnet-deploy/blob/main/src/AWS.Deploy.CLI/Extensions/CustomServiceCollectionExtension.cs) - This serves as the dependency injection container and instantiates all the dependencies required by the deployment tool.
```

Comment:
Replace `instantiates` with `registers`. The services are not instantiated till they are requested.